src/compiler/diagnosticMessages.json

```diff
@@ -1813,5 +1813,9 @@
     "The 'arguments' object cannot be referenced in an arrow function. Consider using a standard function expression.": {
         "category": "Error",
         "code": 9002
+    },
+    "References must appear before any other code.": {
+      "category": "Error",
+      "code": 9003

```

Comment:
please move the error to the 1000 range, as this is a parse error. you can find more information in the [Coding guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#diagnostic-message-codes) section.